e2e/actions/oci.go

```diff
@@ -440,9 +427,23 @@ func (c actionTests) actionOciBinds(t *testing.T) {
 			if !fn(path) {
 				t.Errorf("%s not found on host", path)
 			}
-			if err := os.RemoveAll(path); err != nil {
-				t.Errorf("failed to delete %s: %s", path, err)
-			}
+			// This part needs to be in privileged mode because of the following
```

Comment:
@dtrudg: Please have a look at this comment. This is what was behind the somewhat opaque msgs I sent you on Slack earlier today.

If the behavior described in the comment is what we should be expecting, then it is indeed just a matter of making the e2e test resilient to it, and this should be good to go.

But if this behavior shouldn't be happening (i.e., if the intermediate dirs should be mkdir'd with non-shifted uid/gid, or not mkdir'd at all), then we need to discuss more.

Note also that this is consistent with the behavior of native mode, which also mkdirs missing intermediate directories as needed in this scenario, but there are uid/gid maps in play and they are simply created under the ownership of the user running singularity.